File: android/interactable/src/main/java/com/wix/interactable/physics/PhysicsFrictionBehavior.java
package com.wix.interactable.physics;

import android.graphics.PointF;
import android.util.Log;
import android.view.View;

/**
 * Created by zachik on 12/02/2017.
 */

public class PhysicsFrictionBehavior extends PhysicsBehavior {

    private float friction;
    public PhysicsFrictionBehavior(View target, float friction) {
        super(target, false);
        this.friction = friction;
        this.priority = 2;
    }

    @Override
    public void executeFrameWithDeltaTime(float deltaTime, PhysicsObject physicsObject) {
//        Log.d("InteractableView"," PhysicsFrictionBehavior executeFrameWithDeltaTime friction = " + friction);
        if (!isWithinInfluence()) {
            return;
        }

        float vx = (float) (Math.pow(this.friction, 60.0 * deltaTime) * physicsObject.velocity.x);
        float vy = (float) (Math.pow(this.friction, 60.0 * deltaTime) * physicsObject.velocity.y);

//        Log.d("InteractableView"," vx = " + vx + " cur vx = " + physicsObject.velocity.x);

        physicsObject.velocity = new PointF(vx,vy);
    }
}

File: lib/src/PhysicsFrictionBehavior.ts
import { PhysicsBehavior } from "./PhysicsBehavior";
import { PhysicsObject } from "./PhysicsObject";
import { ITarget } from "./types";

export class PhysicsFrictionBehavior extends PhysicsBehavior {
    friction: number;

    initWithTarget(target: ITarget) {
        super.initWithTarget(target);
        this.priority = 2;
        this.friction = 0.7;
        return this;
    }

    executeFrameWithDeltaTime(deltaTime: number, object: PhysicsObject) {
        if (!this.isWithinInfluence()) {
            return;
        }

        const vx = Math.pow(this.friction, 60.0 * deltaTime) * object.velocity.x;
        const vy = Math.pow(this.friction, 60.0 * deltaTime) * object.velocity.y;

        object.velocity = { x: vx, y: vy };
    }
}